Model/GroupInterface.php

```diff
@@ -11,9 +11,13 @@
 
 namespace FOS\UserBundle\Model;
 
+@trigger_error('Using Groups is deprecated since version 2.2 and will be removed in 3.0.', E_USER_DEPRECATED);
```

Comment:
this should only be in the deprecated class constructor as it triggers deprecation even if we don't use the group class

Reply:
It's more than that. The `GroupableInterface` is also deprecated, and implemented by the main `User` class...
I suggest to revert this PR, it doesn't look ready yet.